Tasks/Common/azure-arm-rest/azure-arm-common.ts

```diff
@@ -67,9 +81,35 @@ export class ApplicationTokenCredentials {
         return this.clientId;
     }
 
-    private getAuthorizationToken(): Q.Promise<string> {
+    private _getMSIAuthorizationToken(): Q.Promise<string> {
         var deferred = Q.defer<string>();
+        let webRequest = new webClient.WebRequest();
+        webRequest.method = "GET";
+        let port = this.msiPort ? this.msiPort : '50342';
```

Comment:
Can we set this value upfront in constructor?

Reply:
Done.